### PR TITLE
executor: support firstrow under new aggregation evaluation framework

### PR DESCRIPTION
**Pull request by XuHuaiyu**
## What have you changed? (mandatory)
support first_row under new aggregation evaluation framework

## What is the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
The existing test cases.

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

no
## Does this PR affect tidb-ansible update? (mandatory)

no

## Does this PR need to be added to the release notes? (mandatory)
no

## Refer to a related PR or issue link (optional)
#6952

### CODE REVIEW

File: executor/aggfuncs/func_first_row.go [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+func (e *firstRow4Int) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup []chunk.Row, pr PartialResult) error {
+	p := (*partialResult4FirstRowInt)(pr)
+	if p.gotFirstRow {
+		return nil
+	}
+	for _, row := range rowsInGroup {
+		input, isNull, err := e.args[0].EvalInt(sctx, row)
+		if err != nil {
+			return errors.Trace(err)
+		}
```
Comment (zz-jason) [inline executor/aggfuncs/func_first_row.go]: set `p.gotFirstRow = true` here, break the calculation after updating the partial result.

File: executor/aggfuncs/func_first_row.go [hunk 2]
```diff
@@ -0,0 +1,474 @@
...
+		if isNull {
+			p.isNull = true
+			break
+		}
+		p.val = input
```
Comment (crazycs520) [inline executor/aggfuncs/func_first_row.go]: break?

**Comment by XuHuaiyu** [comment]
PTAL @zz-jason @crazycs520

File: executor/aggfuncs/func_first_row.go [hunk 3]
```diff
@@ -0,0 +1,483 @@
...
+		if err != nil {
+			return errors.Trace(err)
+		}
+		p.gotFirstRow = true
+		if isNull {
+			p.isNull = true
+			break
+		}
+		// We should copy the input value here to avoid the origin value be covered.
+		p.val = fmt.Sprintf("%s", input)
```
[the remark this answers is not in the document]
Comment (lysu) [reply executor/aggfuncs/func_first_row.go]: https://gist.github.com/lysu/39d4e114632d1790af171ed98ab7d8cb I try to find it too..

File: executor/aggfuncs/func_first_row.go [hunk 4]
```diff
@@ -0,0 +1,439 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggfuncs
+
+import (
+	"fmt"
```
Comment (crazycs520) [inline executor/aggfuncs/func_first_row.go]: imported and not used: "fmt"

**Comment by jackysp** [comment]
Please fix CI first.

**Comment by XuHuaiyu** [comment]
This PR should be merged with https://github.com/pingcap/tidb-test/pull/588 together

**Comment by XuHuaiyu** [comment]
PTAL @zz-jason @crazycs520

File: executor/aggfuncs/func_first_row.go [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+	if p.gotFirstRow {
+		return nil
+	}
+	for _, row := range rowsInGroup {
+		input, isNull, err := e.args[0].EvalInt(sctx, row)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		p.gotFirstRow = true
+		if isNull {
```
Comment (zz-jason) [inline executor/aggfuncs/func_first_row.go]: how about:
```go
p.isNull = isNull
p.val = input
break
```

**Comment by winoros** [comment]
Merge if it's ready @XuHuaiyu

File: executor/aggfuncs/aggfuncs.go [hunk 6]
```diff
@@ -36,6 +36,7 @@ var (
 
 	// All the AggFunc implementations for "SUM" are listed here.
 	// All the AggFunc implementations for "FIRSTROW" are listed here.
+
```
Comment (zz-jason) [inline executor/aggfuncs/aggfuncs.go]: we should put all the implementations for `first_row` in this code block 😂

**Pull request merged**